### PR TITLE
Fixed code quality issues.

### PR DESCRIPTION
**Pull request by lxgr-linux**
- Removed several execs and evals
- Added doc-strings
- De-overwrote built-ins
- Enforced pep guidelines
- Outsourced several Classes
- Fixed security problem with swap_poke

### CODE REVIEW

File: pokete.py [hunk 1]
```diff
@@ -1589,72 +1642,85 @@ def fight_ap_potion(obj, enem, info):
...
                     " < I've never seen one before!", " < Here you go, 200$"])
         if ask_bool(movemap, "Young boy gifted you 200$. Do you want to accept it?"):
             figure.add_money(200)
-        playmap_17.boy_1.will = False
-        used_npcs.append(playmap_17.boy_1.name)
+        npc.will = False
+        used_npcs.append(npc.name)
     else:
-        movemap_text(playmap_17.boy_1.x, playmap_17.boy_1.y,
+        movemap_text(npc.x, npc.y,
```
Comment (MaFeLP) [inline pokete.py:1698]: Unresolved reference to variable `npc`

Comment (lxgr-linux) [reply pokete.py:1698]: @MaFeLP `npc` is defined in line 1689 `npc = ob_maps["playmap_17"].boy_1`.

File: pokete.py [hunk 2]
```diff
@@ -63,6 +66,7 @@ def action(self, ob):
 
 
 class NPC(se.Box):
+    """An NPC ro talk to"""
```
Comment (MaFeLP) [inline pokete.py:69]: Typo
```suggestion
    """An NPC to talk to"""
```

File: pokete.py (hunk 1 above)
Comment (MaFeLP) [reply pokete.py:1724]: PyCharm says otherwise.

Comment (lxgr-linux) [reply pokete.py:1724]: Look into the code then.

Comment (lxgr-linux) [reply pokete.py:1724]: I fixed this

File: pokete.py [hunk 3]
```diff
@@ -572,20 +674,34 @@ def has_item(self, item):
             return False
 
     def remove_item(self, item, amount=1):
+        """Removes a certain amount of an item from the inv"""
         assert amount > 0, "Amounts have to be positive"
         assert item in self.inv, f"Item {name} is not in the inventory"
-        assert self.inv[item]-amount >= 0, f"There are not enought {name}s in the inventory"
+        assert self.inv[item] - amount >= 0, f"There are not enought {name}s in the inventory"
```
Comment (MaFeLP) [inline pokete.py:680]: - Unresolved Reference 'name'
- Also: Typo
```suggestion
        assert self.inv[item] - amount >= 0, f"There are not enough {item}s in the inventory"
```

Comment (lxgr-linux) [reply pokete.py]: Found the same issue in line 679.

Comment (MaFeLP) [reply pokete.py]: Yes. Was currently trying to get a new suggestion going, but gh didn't want to...

Comment (lxgr-linux) [reply pokete.py]: Anyway, I fixed it.

**Comment by MaFeLP** [comment]
Does "Code Quality" also mean grammar and PEP8 are being addressed?

**Comment by lxgr-linux** [comment]
> Does "Code Quality" also mean grammar and PEP8 are being addressed?

Yes, why do you ask?

**Comment by MaFeLP** [comment]
> > Does "Code Quality" also mean grammar and PEP8 are being addressed?
> 
> Yes, why do you ask?

Did a #45 then

**Pull request merged**